components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -710,10 +748,33 @@ protected void doAuthentication(HttpServletRequest request, HttpServletResponse
                 audit.info(String.format(AUDIT_MESSAGE, initiator, "Authenticate", "ApplicationAuthenticationFramework",
                         data, SUCCESS));
             }
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                Map<String, Object> params = new HashMap<>();
+                params.put(FrameworkConstants.LogConstants.STEP, stepConfig.getOrder());
+                params.put(FrameworkConstants.LogConstants.SERVICE_PROVIDER, context.getServiceProviderName());
+                params.put(FrameworkConstants.LogConstants.TENANT_DOMAIN, context.getTenantDomain());
+                params.put(FrameworkConstants.LogConstants.IDP, stepConfig.getAuthenticatedIdP());
+                params.put(FrameworkConstants.LogConstants.AUTHENTICATOR_NAME,
+                        stepConfig.getAuthenticatedAutenticator().getName());
+                LoggerUtils.triggerDiagnosticLogEvent(FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, params,
+                        LogConstants.SUCCESS, "Authentication success for step: " + stepConfig.getOrder(),
+                        FrameworkConstants.LogConstants.ActionIDs.HANDLE_AUTH_STEP, null);
+            }
         } catch (InvalidCredentialsException e) {
             if (LOG.isDebugEnabled()) {
                 LOG.debug("A login attempt was failed due to invalid credentials", e);
             }
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                 Map<String, Object> params = getContextParamsForDiagnosticLogs(context, authenticatorConfig,
+                        stepConfig);
+                Optional.of(e.getUser()).ifPresent(user -> {
+                    params.put(FrameworkConstants.LogConstants.USER, user.getLoggableUserId());
+                    params.put(FrameworkConstants.LogConstants.USER_STORE_DOMAIN, user.getUserStoreDomain());
+                });
+                LoggerUtils.triggerDiagnosticLogEvent(FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, params,
+                        LogConstants.FAILED, "Authentication failed: " + e.getMessage(),
```

Comment:
can we say with which authenticator/idp?

Reply:
+1. Added with https://github.com/wso2/carbon-identity-framework/pull/4326/commits/9a1423aa5dca4a529cd5a0abe0c755ac81eee53c